x/liquidity/types/swap.go

```diff
@@ -507,8 +507,13 @@ func FindOrderMatch(direction OrderDirection, swapMsgStates []*SwapMsgState, exe
 						TransactedCoinAmt: offerAmt.Mul(fractionalMatchRatio).Ceil(),
 						SwapMsgState:      matchOrder,
 					}
-					// Fee, Exchanged amount are values that should not be overmeasured, so it is lowered conservatively considering the decimal error.
-					matchResult.OfferCoinFeeAmt = matchResult.SwapMsgState.ReservedOfferCoinFee.Amount.ToDec().Mul(fractionalMatchRatio)
+					if matchResult.OfferCoinAmt.Sub(matchResult.TransactedCoinAmt).LTE(sdk.OneDec()) {
+						// If OfferCoinAmt and TransactedCoinAmt are close, use ReservedOfferCoinFee immediately to avoid decimal errors.
+						matchResult.OfferCoinFeeAmt = matchResult.SwapMsgState.ReservedOfferCoinFee.Amount.ToDec()
+					} else {
+						// Fee, Exchanged amount are values that should not be overmeasured, so it is lowered conservatively considering the decimal error.
+						matchResult.OfferCoinFeeAmt = matchResult.SwapMsgState.ReservedOfferCoinFee.Amount.ToDec().Mul(fractionalMatchRatio).Ceil()
```

Comment:
I don't understand the comment for this line of code. Can you elaborate this more in detail? The way I read the code is that it multiplies `ReservedOfferCoinFee` amount with `fractionalMatchRatio` and ceil it for fractional match.